components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/utils/RestApiUtil.java

```diff
@@ -1633,23 +1633,29 @@ public static boolean checkIfAnonymousAPI(Message inMessage) {
     /**
      * This method is used to get the scope list from the yaml file
      *
-     * @param fileName file name of YAML
-     * @return MAP of scope list for a portal
+     * @return MAP of scope list for all portal
      */
-    public static synchronized Map<String,String> getScopes(String fileName) throws APIManagementException {
-        String definition = null;
-        try {
-            definition = IOUtils
-                    .toString(RestApiUtil.class.getResourceAsStream(fileName), "UTF-8");
-        } catch (IOException  e) {
-            throw new APIManagementException("Error while reading the swagger definition ,",
-                    ExceptionCodes.DEFINITION_EXCEPTION);
-        }
-        APIDefinition oasParser = OASParserUtil.getOASParser(definition);
-        Set<Scope> scopeSet = oasParser.getScopes(definition);
-        Map<String,String> portalScopeList = new HashMap<String, String>();
-        for (Scope entry : scopeSet) {
-            portalScopeList.put(entry.getName(),entry.getDescription());
+    public static  Map<String, List<String>> getScopesRole() throws APIManagementException {
```

Comment:
Shall we rename this method to getScopesInfoFromAPIYamlDefinitions?

Reply:
Done with b285a8a